students/cjfortu/lesson09/mailroom_00/donor_models.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python
+
+"""
+The Mailroom assignment, donor models.
+
+Donor object creation is under the Donor class, while assembling the donor objects is in the DonorCollection class.
+The donor object collection structure is:
+{'name": object'}
+In words, it is a dict with each donor name in string form as the keys, and the donor object as the value.
+
+The remaining features follow the assignment instructions.
+"""
+
+
+class Donor:
+    """Class that holds and accesses information about a single donor."""
+    new_donations = []
+    
+    def __init__(self, donor_name=None, *donations):
+        """Initiate a donor by name and donations."""
+        self.name = donor_name
+        self.donations = [*donations]
+        self.new_donations = [*donations]
+
+    @property
+    def number_donations(self):
+        return len(self.donations)
+    
+    @property
+    def lifetime_donations_sum(self):
+        return sum(self.donations)
+    
+    @property
+    def average_donations(self):
+        return self.lifetime_donations_sum / self.number_donations
+        
+    def add_donations(self, new_donations):
+        """Append new donations to the existing donations attribute."""
+        [self.donations.append(donation) for donation in new_donations]
+        self.new_donations = new_donations
+        
+    @property
+    def compose_letter(self):
+        """Generate text for a single letter."""
+        text = "\n\nHi {},\n\nThank you for your total donation of ${:,.2f}.\n\n\nVR\n\nThe Mailroom\n(555) 555-5555".\
+            format(self.name, sum(self.new_donations))
```

Comment:
you should only print the most recent donation here `self.donations[-1]` should satisfy this.

Reply:
This goes with my comment above about **donations** vs **new_donations**